### PR TITLE
Added unit tests for lib/markdown

### PR DESCRIPTION
**Pull request by rayou**
Added unit tests for the issues in recent releases.

### CODE REVIEW

File: browser/lib/consts.js [hunk 1]
```diff
@@ -1,10 +1,9 @@
 const path = require('path')
 const fs = require('sander')
 const { remote } = require('electron')
-const { app } = remote
```
Comment (rayou) [inline browser/lib/consts.js]: I couldn't make `app` extracted from `remote` while running tests, following error throws.
```bash
var app = remote.app;
                ^

TypeError: Cannot read property 'app' of undefined
```
Any ideas why this happened?

File: package.json [hunk 2]
```diff
@@ -92,7 +92,7 @@
     "uuid": "^3.2.1"
   },
   "devDependencies": {
-    "ava": "^0.16.0",
+    "ava": "^0.25.0",
```
Comment (rayou) [inline package.json]: updated to 0.25.0 for snapshot support

File: tests/lib/rc-parser-test.js [hunk 3]
```diff
@@ -29,5 +29,5 @@ test('RcParser should return a json object', t => {
 })
 
 function filePath (filename) {
-  return path.join('boostnoterc', filename)
+  return path.join(`${__dirname}/boostnoterc`, filename)
```
Comment (rayou) [inline tests/lib/rc-parser-test.js]: breaking change in ava [v0.17.0](https://github.com/avajs/ava/releases/tag/v0.17.0)

File: browser/lib/consts.js (hunk 1 above)
Comment (Rokt33r) [reply browser/lib/consts.js]: I assume we have to make `remote` Mockable.

File: tests/helpers/setup-electron-mock.js [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import mock from 'mock-require'
+
+const noop = () => {}
+
+mock('electron', {
```
Comment (rayou) [inline tests/helpers/setup-electron-mock.js]: added mok electron.

File: browser/lib/consts.js (hunk 1 above)
Comment (rayou) [reply browser/lib/consts.js]: As result, `require('electron')` just return the path of electron binary, a string.

I think it's an good idea to mock `electron`, so I created a new mock helper in `tests/helpers/setup-electron-mock.js`, we can add more mocked function into it if needed.

**Comment by rayou** [comment]
As result, `require('electron')` just return the path of electron binary, a string.

I think it's an good idea to mock `electron`, so I created a new mock helper in `tests/helpers/setup-electron-mock.js`, we can add more mocked function into it if needed.

The change in `browser/lib/consts.js` has been reverted.

**Comment by Rokt33r** [comment]
Great work!! 👍 👍 👍

**Comment by rayou** [comment]
fixed conflicts.

**Pull request merged**